src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/CatalogPromotion.xml

```diff
@@ -0,0 +1,189 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+
+<!--
+
+ This file is part of the Sylius package.
+
+ (c) Sylius Sp. z o.o.
+
+ For the full copyright and license information, please view the LICENSE
+ file that was distributed with this source code.
+
+-->
+
+<resources
+    xmlns="https://api-platform.com/schema/metadata/resources-3.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="https://api-platform.com/schema/metadata/resources-3.0 https://api-platform.com/schema/metadata/resources-3.0.xsd"
+>
+    <resource class="Sylius\Component\Core\Model\CatalogPromotion" shortName="CatalogPromotion">
+        <validationContext>
+            <values>
+                <value name="groups">
+                    <values>
+                        <value>sylius</value>
+                    </values>
+                </value>
+            </values>
+        </validationContext>
+
+        <operations>
+            <operation class="ApiPlatform\Metadata\Get"  uriTemplate="/admin/catalog-promotions/{code}.{_format}">
+                <normalizationContext>
+                    <values>
+                        <value name="groups">
+                            <values>
+                                <value>admin:catalog_promotion:read</value>
+                            </values>
+                        </value>
+                    </values>
+                </normalizationContext>
+            </operation>
+
+            <operation class="ApiPlatform\Metadata\GetCollection" uriTemplate="/admin/catalog-promotions.{_format}">
+                <normalizationContext>
+                    <values>
+                        <value name="groups">
+                            <values>
+                                <value>admin:catalog_promotion:read</value>
+                            </values>
+                        </value>
+                    </values>
+                </normalizationContext>
+                <filters>
+                    <filter>sylius.api.catalog_promotion_search_filter</filter>
+                    <filter>sylius.api.catalog_promotion_enabled_filter</filter>
+                    <filter>sylius.api.catalog_promotion_start_date_filter</filter>
+                    <filter>sylius.api.catalog_promotion_end_date_filter</filter>
+                    <filter>Sylius\Bundle\ApiBundle\Filter\Doctrine\CatalogPromotionChannelFilter</filter>
```

Comment:
Would be cool to unify services definitions (I assume it's services) either to string or FQCN.
```suggestion
                    <filter>sylius.api.catalog_promotion_channel_filter</filter>
```

Reply:
Filters created in Sylius have an ID as FQCN, while those using filters from APIP have a string. We will see what others have to say.